packages/central-server/src/apiV2/import/importSurveys/importSurveys.js

```diff
@@ -417,11 +422,23 @@ export async function importSurveys(req, res) {
   respond(res, { message: 'Imported surveys' });
 }
 
-async function processOptionSetName(models, name) {
+async function processOptionSetName(models, name, excelRowNumber, tabName) {
   // TODO: Figure out why undefined value is returning all results.
   if (name) {
-    const optionSet = await models.optionSet.findOne({ name });
-    return optionSet.id;
+    const constructImportValidationError = (message, field) =>
+      new ImportValidationError(message, excelRowNumber, field, tabName);
+    try {
+      const optionSet = await models.optionSet.findOne({ name });
+      return optionSet.id;
+    } catch (error) {
+      console.log(error.message);
+      const message =
+        error.message === "Cannot read property 'id' of null"
+          ? 'The option set listed cannot be found in the database'
+          : error.message;
```

Comment:
@chris-pollard Do you want to locate this error where trying to read 'id' from a null record? Maybe we could throw a better generic error message from there?

Reply:
My bad. The `Cannot read property 'id' of null` error is not from database model, instead it is from reading id from null on line 432.
We could throw an error by checking if return result is null.

Reply:
Hey @biaoli0 fixed it up, thanks